test/scale/scale_tester.go

```diff
@@ -178,94 +186,117 @@ scaleInTest:
 }
 
 func (t *PodSchedulingOnFailoverScaleTest) Run(client clientset.Interface, namespace *v1.Namespace, schedulerName string) {
-	statefulSets := []*resources.TestStatefulset{}
+	totalNumberOfPods := t.Replicas
+
+	nodes := nodeutil.ListAgentNodeNames(client, t.Pod.IsWindows)
+	skipper.SkipUnlessAtLeast(len(nodes), 2, "Need at least 2 agent nodes to verify the test case.")
+	midNode := len(nodes) / 2
+
+	testPod := resources.TestPod{
+		Client: client,
+	}
 
-	ginkgo.By("Initiating statefulset setup")
+	// uncordon the first half nodes
+	for _, node := range nodes[:midNode] {
+		node := node
+		framework.Logf("Uncordoning %s.", node)
+		testPod.SetNodeUnschedulable(node, false)
+	}
+
+	// cordon the second half nodes)
+	for _, node := range nodes[midNode:] {
+		node := node
+		framework.Logf("Cordoning %s.", node)
+		testPod.SetNodeUnschedulable(node, true)
+	}
 
+	// start scale out test
+	startScaleOut := time.Now()
```

Comment:
This is wrong in the other scale test as well, but I think we should start measuring the scale out right before `statefulSet.CreateWithoutWaiting()`

Reply:
Make sense, will change the other sale test together.